File: ResourceManager/ViewModels/Notes/NotePageViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Security
Imports ResourceManager.Configuration
Imports ResourceManager.Helpers
Imports System.Collections.ObjectModel
Imports System.Linq
Imports Microsoft.Extensions.DependencyInjection

Namespace ResourceManager.ViewModels
    Public Class NotePageViewModel
        Inherits PageViewModel
        Public Sub New()

            PageTitle = "Note"
            PageIcon = AssetManager.Instance.GetImage("Note.png")
            PageColor = NotePageColor
            MyBase.HasPermission = AuthManager.User.IsInPermission(VIEW_NOTE_PERMISSION)

            Dim tagsContent = Application.ServiceProvider.GetRequiredService(Of TagsContentViewModel)()
            tagsContent.PageColor = PageColor
            tagsContent.NoteColumnWidth = 110

            MenuItems = New ObservableCollection(Of PageContentViewModel) From {
    Application.ServiceProvider.GetRequiredService(Of NoteContentViewModel)(),
    tagsContent
}

            SelectedItem = MenuItems.FirstOrDefault()
        End Sub
    End Class
End Namespace

File: ResourceManager/ViewModels/Notes/NotePageViewModel.cs
using Microsoft.Extensions.DependencyInjection;
using NetCore.Mvvm.Abstractions;
using NetCore.Security;
using ResourceManager.Configuration;
using ResourceManager.Helpers;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace ResourceManager.ViewModels
{
    public class NotePageViewModel : PageViewModel
    {
        public NotePageViewModel()
        {

            PageTitle = "Note";
            PageIcon = AssetManager.Instance.GetImage("Note.png");
            PageColor = Constants.NotePageColor;
            HasPermission = AuthManager.User.IsInPermission(Constants.VIEW_NOTE_PERMISSION);

            var tagsContent = App.ServiceProvider.GetRequiredService<TagsContentViewModel>();
            tagsContent.PageColor = PageColor;
            tagsContent.NoteColumnWidth = 110;

            MenuItems = new ObservableCollection<PageContentViewModel>
            {
                App.ServiceProvider.GetRequiredService<NoteContentViewModel>(),
                tagsContent
            };

            SelectedItem = MenuItems.FirstOrDefault();
        }
    }
}
